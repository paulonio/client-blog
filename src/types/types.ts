export interface Post {
  username: string;
  date: string;
  title: string;
  body: string;
  urlToImage: string;
}
