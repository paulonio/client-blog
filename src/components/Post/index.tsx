import React, { FC } from 'react';
import type { PostType } from '@/types/types';

import styles from './styled.module.scss';

interface PostProps {
  post: PostType;
}

const Post: FC<PostProps> = ({ post }) => {
  const { tag, title, body, urlToImage } = post;
  return (
    <div className={styles.post}>
      <div className={styles.image}>
        <img src={urlToImage} alt={`${tag} post`} />
      </div>
      <div className={styles.content}>
        <h5>{tag}</h5>
        <h2>{title}</h2>
        <p className="body-s">{body}</p>
      </div>
    </div>
  );
};

export default Post;
