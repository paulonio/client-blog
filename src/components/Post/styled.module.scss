@import '../../styles/sass/constants';

.post {
  width: 100%;
  max-width: 1146px;
  display: flex;
  align-items: center;
  gap: 32px;
}

.image {
  width: 100%;
  max-width: 490px;
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: hidden;
}

.content {
  display: flex;
  flex-direction: column;
  align-items: start;
  row-gap: 16px;

  & h5 {
    margin-bottom: 4px;
    color: $color-secondary;
  }
}