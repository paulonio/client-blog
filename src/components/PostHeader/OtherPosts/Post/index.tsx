import React, { FC } from 'react';
import Link from 'next/link';
import { useTranslation } from 'next-i18next';

import type { PostType } from '@/types/types';
import { sen } from '@/styles/fonts';

import { stringToKey } from '@/utils/utils';
import styles from '@/components/PostHeader/styled.module.scss';

interface PostProps {
  post: PostType;
}

const Post: FC<PostProps> = ({ post }) => {
  const { t } = useTranslation(['blog', 'common']);
  const { id, username, date, title, body, urlToImage } = post;

  return (
    <Link className={styles.post__wrapper} href={`/blog/${id}`}>
      <div className={styles.post__image}>
        <img src={urlToImage} alt={`${username} post`} />
      </div>
      <div className={styles.post__content}>
        <p className="body-l">
          {t('by', { ns: 'common' })}{' '}
          <span className={styles.username}>{t(stringToKey(username), { ns: 'common' })}</span> |{' '}
          {t(stringToKey(date), { ns: 'common' })}
        </p>
        <h3 className={sen.className}>{t(stringToKey(title))}</h3>
        <p className="body-s">{t(stringToKey(body))}</p>
      </div>
    </Link>
  );
};

export default Post;
