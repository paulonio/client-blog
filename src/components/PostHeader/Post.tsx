import React, { FC } from 'react';
import { PostType } from '@/types/types';

import styles from './styled.module.scss';
import { sen } from '@/styles/fonts';

interface PostProps {
  post: PostType;
}

const Post: FC<PostProps> = ({ post }) => {
  const { username, date, title, body, urlToImage } = post;

  return (
    <div className={styles.post__wrapper}>
      <div className={styles.post__image}>
        <img src={urlToImage} alt={`${username} post`} />
      </div>
      <div className={styles.post__content}>
        <p className="body-l">
          By <span className={styles.username}>{username}</span> | {date}
        </p>
        <h3 className={sen.className}>{title}</h3>
        <p className="body-s">{body}</p>
      </div>
    </div>
  );
};

export default Post;
