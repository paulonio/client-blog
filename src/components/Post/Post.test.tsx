import React from 'react';
import { render } from '@testing-library/react';
import Post from '.';
import { PostType } from '@/types/types';

const TEST_DATA: PostType = {
  id: 'test',
  body: 'test',
  category: 'test',
  date: 'test',
  tags: ['test'],
  title: 'test',
  urlToAvatar: 'test',
  urlToImage: 'test',
  username: 'test',
};

describe('Post', () => {
  it('should render Post', () => {
    const { container } = render(<Post post={TEST_DATA} />);

    expect(container).toMatchSnapshot();
  });
});
