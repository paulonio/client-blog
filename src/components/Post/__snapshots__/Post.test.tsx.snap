// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render Post 1`] = `
<div>
  <div
    class="post"
  >
    <div
      class="image"
    >
      <img
        alt="test post"
        src="test"
      />
    </div>
    <div
      class="content"
    >
      <h5>
        test
      </h5>
      <h2>
        test
      </h2>
      <p
        class="body-s"
      >
        test
      </p>
    </div>
  </div>
</div>
`;
